docs/backup-restore-integration.md

```diff
@@ -0,0 +1,334 @@
+# Backing Up and Restoring KubeVirt resources
+
+## Introduction
+
+This document is intended for developers building backup/disaster recovery solutions for KubeVirt.
+
+Such solutions typically perform the following high level operations.
+
+### Backup
+
+1.  Build dependency graph for all required Kubernetes resources
+2.  Quiesce (freeze) applications
+3.  Snapshot PersistentVolumeClaim data
+4.  Unquiesce (thaw) applications
+5.  Copy all necessary Kubernetes resource definitions to a shared storage location
+6.  (optional) Export snapshotted PVC data to a shared storage location
+
+Steps 3, 5, and 6 are beyond the scope of this document.
+
+### Restore
+
+1.  Populate PersistentVolumeClaims with snapshot data
+2.  Sanitize and apply all relevant Kubernetes resource definitions
+
+Step 1 is beyond the scope of this document.
+
+## Existing KubeVirt Backup Solutions
+
+### Velero Plugin
+
+[Velero](https://velero.io/) is a popular tool for backing up/migrating Kubernetes clusters.  The KubeVirt team actively maintains a [plugin](https://github.com/kubevirt/kubevirt-velero-plugin) for use with Velero.  The plugin implements much of the logic described in this document.
+
+### VirtualMachineSnapshot + VirtualMachineExport API
+
+The [VirtualMachineSnapshot API](https://kubevirt.io/user-guide/operations/snapshot_restore_api/) provides an easy way for KubeVirt users to backup VirtualMachines within a cluster.  On its own, it is not suitable for offsite backup or disaster recovery.  But combined with the [VirtualMachineExport API](https://kubevirt.io/user-guide/operations/export_api/), VirtualMachine volumes may be made available for copy to remote locations.
+
+## Building the KubeVirt Object Graph
+
+![Object Graph Example](backup-graph.png "VM Graph")
+
+In this section, KubeVirt resources and their relationships will be explored by showing how yaml snippets map to nodes in an object graph.
+
+Nodes in the object graph are represented by the following tuple:
+
+(APIGroup, Kind, namespace, name)
+
+### VirtualMachine Object Graph
+
+```yaml
+apiVersion: kubevirt.io/v1
+kind: VirtualMachine
+metadata:
+  name: vm1
+  namespace: ns1
+...
+```
+
+- ("kubevirt.io", "VirtualMachine", "ns1", "vm1")
+
+#### spec.instancetype
+
+```yaml
+...
+spec:
+  instancetype:
+    kind: VirtualMachineInstancetype
+    name: small
+...
+```
+
+- ("kubevirt.io", "VirtualMachineInstancetype", "ns1", "small")
```

Comment:
isnt the group here instancetype.kubevirt.io ?

Reply:
yup